app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -571,12 +571,21 @@ class BrowserRobot {
     }
 
     fun fillAndSubmitLoginCredentials(userName: String, password: String) {
-        userNameTextBox.setText(userName)
-        passwordTextBox.setText(password)
+        var currentTries = 0
+        while (currentTries++ < 3) {
+            try {
+                mDevice.waitForIdle(waitingTime)
+                userNameTextBox.setText(userName)
+                passwordTextBox.setText(password)
+                submitLoginButton.click()
 
-        submitLoginButton.click()
+                mDevice.waitForObjects(mDevice.findObject(UiSelector().resourceId("$packageName:id/save_confirm")))
 
-        mDevice.waitForObjects(mDevice.findObject(UiSelector().resourceId("$packageName:id/save_confirm")))
+                break
+            } catch (e: UiObjectNotFoundException) {
+                println("Failed to find locator: ${e.localizedMessage}")
```

Comment:
Swap this to `Log.e` and we can use a `BROWSER_ROBOT` tag
```Log.e("BROWSER_ROBOT", "Failed to find locator: ${e.localizedMessage}")```

---

I'm surprised we don't have to reload the page again? Maybe you tried that.

Reply:
Done, will keep in mind to log the error using the corresponding robot as a tag.

> I'm surprised we don't have to reload the page again? Maybe you tried that.

In this case, we fill and submit the login credentials in a PWA which can't be reloaded.
We can only close and re-open the PWA.